From e1c93ef4696d03a79cc3a6d6451647f7657b8d01 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Sat, 12 Oct 2024 01:22:07 +0530
Subject: [PATCH] meson: Use the specified python if it's custom

This allows using a python wrapper for cross-compiling introspection
for macOS x86_64 on macOS arm64. The python interpreter needs to load
x86_64 modules, so it must be run with `arch -x86_64` since it is
a fat binary with both archs in it.

The only reason why we'd need to use python3.X is for the Linux distro
use-case, and in that case python is in /usr/bin.
---
 tools/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/meson.build b/tools/meson.build
index 7237a0b6..f9625356 100644
--- a/tools/meson.build
+++ b/tools/meson.build
@@ -9,7 +9,7 @@ if with_doctool
   tools += [['g-ir-doc-tool', 'docmain', 'doc_main']]
 endif
 
-if cc.get_id() == 'msvc'
+if not python.path().startswith('/usr/bin')
   python_cmd = '/usr/bin/env ' + python.path()
 else
   python_cmd = '/usr/bin/env python@0@'.format(python.language_version().split('.')[0])
-- 
2.45.2

